.github/workflows/action-integration.yml

```diff
@@ -4,14 +4,16 @@ on:
   push:
     branches:
       - main
-  pull_request:
+  pull_request_target:
 jobs:
   action-integration-testing:
     name: Action Integration Testing
     runs-on: ubuntu-latest
     steps:
-      - uses: actions/checkout@v3
-        name: Checkout
+      - name: Checkout
+        uses: actions/checkout@v4.1.6
+        with:
+          ref: ${{ github.event.pull_request.head.sha }}
```

Comment:
I just tested this and this breaks any running of this test on non-prs. This test runs on every push to main too.

Reply:
You could add a coalesce with GitHub.ref